### ISSUE TITLE
`deno bundle` not working with imports

### ISSUE THREAD

**Issue by davidbailey00** [denoland/deno#4206]
## Steps to reproduce

Create `server.js` with the following content:

```js
import { serve } from "https://deno.land/std@v0.35.0/http/server.ts";
const s = serve({ port: 8000 });
console.log("http://localhost:8000/");
for await (const req of s) {
  req.respond({ body: "Hello World\n" });
}

// this is here to work around #4207
await console.log("test");
```

Run `deno bundle server.js out.js`

Run `deno out.js`

## Expected result

The code is bundled correctly and runs

## Actual result

```sh
$ deno bundle server.js out.js
Bundling "file:///path/to/server.js"
Emitting bundle to "out.js"
2.7 kB emitted.
$ deno out.js
error: Uncaught TypeError: Cannot destructure property 'd' of 'mMap.get(...)' as it is undefined.
► file:///path/to/out.js:45:17
```

The bundle doesn't seem to be complete - it's missing the HTTP library.

## Notes

This seems related to #4031 but adding `export {};` to the file doesn't fix the issue.

```
deno 0.35.0
v8 8.1.310
typescript 3.8.2
```

**Comment by ry** [comment]
cc @kitsonk

**Comment by nayeemrmn** [comment]
Better repro:
`a.js`:
```js
export const a = 1;
```
`b.js`:
```js
import { a } from "./temp3.js";
a; // Need to reference import.
```

`deno bundle b.js b.bundle.js && deno b.bundle.js`:
```
Bundling "file:///mnt/c/Users/Nayeem/projects/deno/b.js"
Emitting bundle to "b.bundle.js"
2.4 kB emitted.
```
```
error: Uncaught TypeError: Cannot destructure property 'd' of 'mMap.get(...)' as it is undefined.
► file:///mnt/c/Users/Nayeem/projects/deno/b.bundle.js:45:17

45         const { d } = mMap.get(id);
                   ^

    at enq (file:///mnt/c/Users/Nayeem/projects/deno/b.bundle.js:45:17)
    at enq (file:///mnt/c/Users/Nayeem/projects/deno/b.bundle.js:47:9)
    at __inst_s (file:///mnt/c/Users/Nayeem/projects/deno/b.bundle.js:95:5)
    at file:///mnt/c/Users/Nayeem/projects/deno/b.bundle.js:117:1
```

**Comment by kitsonk** [comment]
Taking a look... 😕

**Comment by kitsonk** [comment]
This is related analysis of dependencies in JavaScript (related to #4040 and #4120)... Fix incoming.

**Comment by kitsonk** [comment]
I think this was closed by 83d902a7803adb0c69fe2c98e692a50dae446db9.  While it contains a top level `for...await`, there is #4207 for tracking that.  There _was_ an issue with bundling JavaScript files even without a top level `for...await`.
